tests/test_dataclasses.py

```diff
@@ -1841,6 +1841,20 @@ class Child(Parent):
     assert child.y == 1
 
 
+def test_kw_only_inheritance_on_field() -> None:
```

Comment:
This would previously fail with an unhandled exception with 3.9. Although it doesn't make sense to use it in 3.9 (and we could add a warning -- not worth the effort as we'll drop support for it soon), still better to not hard error.